### ISSUE TITLE
SIGSEGV with 32 bit Stalker on 64 bit Android

### ISSUE THREAD

**Issue by manizzle** [frida/frida#238]
I am trying to run Stalker on a 64 bit version of android. I try to run the following python/JS code on the Google Photos application and get errors like this:

```
02-21 15:39:21.505 22953-23003/? E/linker: readlink('/proc/self/fd/51') failed: Permission denied [fd=51]
02-21 15:39:21.506 22953-23003/? E/linker: warning: unable to get realpath for the library "/data/local/tmp/re.frida.server/frida-agent-32.so". Will use given name.
02-21 15:39:21.573 22953-23003/? A/libc: Fatal signal 11 (SIGSEGV), code 2, fault addr 0x45913500 in tid 23003 (oid.apps.photos)
02-21 15:39:21.678 606-606/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
02-21 15:39:21.679 606-606/? A/DEBUG: Build fingerprint: 'google/angler/angler:6.0/MDA89D/2294819:user/release-keys'
02-21 15:39:21.679 606-606/? A/DEBUG: Revision: '0'
02-21 15:39:21.679 606-606/? A/DEBUG: ABI: 'arm'
02-21 15:39:21.680 606-606/? A/DEBUG: pid: 22953, tid: 23003, name: oid.apps.photos  >>> com.google.android.apps.photos <<<
02-21 15:39:21.680 606-606/? A/DEBUG: signal 11 (SIGSEGV), code 2 (SEGV_ACCERR), fault addr 0x45913500
02-21 15:39:21.716 606-606/? A/DEBUG:     r0 d8b8412d  r1 459134fc  r2 00000001  r3 ffffffff
02-21 15:39:21.717 606-606/? A/DEBUG:     r4 da78f4fc  r5 000000c2  r6 da78f4fc  r7 d8b8412c
02-21 15:39:21.717 606-606/? A/DEBUG:     r8 00000003  r9 d8b84158  sl da78f4fc  fp d8b8412c
02-21 15:39:21.717 606-606/? A/DEBUG:     ip 0000000b  sp da78ef7c  lr 00000000  pc d9e79254  cpsr 800b0030
02-21 15:39:21.718 606-606/? A/DEBUG: backtrace:
02-21 15:39:21.719 606-606/? A/DEBUG:     #00 pc 002b4254  /data/local/tmp/re.frida.server/frida-agent-32.so (offset 0x64000)
02-21 15:39:21.719 606-606/? A/DEBUG:     #01 pc 00000000  <unknown>
02-21 15:39:22.059 606-606/? A/DEBUG: Tombstone written to: /data/tombstones/tombstone_06
02-21 15:39:22.059 606-606/? E/DEBUG: AM write failed: Broken pipe
```

```
import frida, sys


with open("inject2.js") as fp:
	jscode = fp.read() 

def on_message(message, data):
	print message, data

aid2 = "com.google.android.apps.photos"
process = frida.get_usb_device().attach(aid2)
process.enable_jit()


script = process.create_script(jscode)
script.on("message", on_message)
script.load()
sys.stdin.read()
```

```
"use strict";

console.log("hi");
Stalker.follow(22953, {
	events: {
		call: true,
		ret: false,
		exec: false
	},
	onReceive : function (events) {
		console.log("wooo");
	}
});
```

I get the same error when I use ```Process.getCurrentThreadId()```

**Comment by max01203** [comment]
Hi,
Did you use the arm-64 version of the frida-server?

Thanks,

**Comment by manizzle** [comment]
Yeah I did

**Comment by iddoeldor** [comment]
Solved ?

**Comment by oleavr** [comment]
Fixed in Frida 12.9 – Stalker is finally available on ARM32.

**Comment by manizzle** [comment]
wow awesome!
